File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/library/repository/permissions/storage/read/LibraryStorageReadPermissionsRequirementsProvider.java
package com.lasthopesoftware.bluewater.client.browsing.library.repository.permissions.storage.read;

import androidx.annotation.NonNull;

import com.lasthopesoftware.bluewater.client.browsing.library.repository.Library;

public final class LibraryStorageReadPermissionsRequirementsProvider implements ILibraryStorageReadPermissionsRequirementsProvider {

	@Override
	public boolean isReadPermissionsRequiredForLibrary(@NonNull Library library) {
		return library.isUsingExistingFiles() || Library.SyncedFileLocation.ExternalDiskAccessSyncLocations.contains(library.getSyncedFileLocation());
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/library/repository/permissions/storage/read/LibraryStorageReadPermissionsRequirementsProvider.kt
package com.lasthopesoftware.bluewater.client.browsing.library.repository.permissions.storage.read

import com.lasthopesoftware.bluewater.client.browsing.library.repository.Library

object LibraryStorageReadPermissionsRequirementsProvider :
    CheckLibraryStorageReadPermissionRequirements {
    override fun isReadPermissionsRequiredForLibrary(library: Library): Boolean {
        return library.isUsingExistingFiles || Library.SyncedFileLocation.ExternalDiskAccessSyncLocations.contains(
            library.syncedFileLocation
        )
    }
}
